internal/testcontainersdocker/docker_socket.go

```diff
@@ -11,3 +18,24 @@ var DockerSocketPathWithSchema = DockerSocketSchema + DockerSocketPath
 
 // TCPSchema is the tcp schema.
 var TCPSchema = "tcp://"
+
+func init() {
```

Comment:
I had a local branch doing something similar, but creating a `docker_socket_windows.go` file, directly assigning the values we need:

```go
// DockerSocketSchema is the npipe schema.
var DockerSocketSchema = "npipe://"

// DockerSocketPath is the path to the docker socket under Windows systems.
var DockerSocketPath = "//./pipe/docker_engine"
```

I'm not sure if we should use this simpler approach, or instead delegate to the docker client as in this PR. What are your thoughts on this?

Reply:
That would've solved it for sure! 🛬

I'm in favour of delegating to the Docker library because it honours their responsibility, in my mind.

However, you've already chosen to use a hard-coded value, therefore I'd support your approach blindly 😎 you definitely know best.

Let me know how you'd like to proceed. I'm available to make the desired changes.

Reply:
>I'm in favour of delegating to the Docker library because it honours their responsibility, in my mind.

I like this too. I think we can stick to it. Thanks for your support!